Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.Commands.cpp

```diff
@@ -36,10 +48,28 @@ int CLuaFunctionDefs::AddCommandHandler ( lua_State* luaVM )
         CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine ( luaVM );
         if ( pLuaMain )
         {
-            // Add them to our list over command handlers
-            if ( m_pRegisteredCommands->AddCommand ( pLuaMain, strKey, iLuaFunction, bCaseSensitive ) )
+            if ( bType )
             {
-                lua_pushboolean ( luaVM, true );
+                // Add them to our list over command handlers
+                if ( strKey == "" || m_pRegisteredCommands->AddCommand(pLuaMain, strKey, iLuaFunction, bCaseSensitive) )
+                {
+                    lua_pushboolean( luaVM, true );
+                    return 1;
+                }
+            }
+            else
+            {
+                // Add them to our list over command handlers from table {"command1","command2",...}
+                bool bCommandError = false;
+                for (uint i = 0; i < vecKey.size(); i++)
+                {
+                    if ( vecKey[i] == "" || !m_pRegisteredCommands->AddCommand(pLuaMain, vecKey[i], iLuaFunction, bCaseSensitive) )
+                    {
+                        bCommandError = true;
+                        break;
```

Comment:
You have to keep track of already added commands and remove these in case of an error. Suppose you call the function with a duplicate: `addCommandHandler({"foobar", "foobar"}, ...)` you will end up receiving **false**, but the command has been added anyway.